src/components/Provider.js

```diff
@@ -20,7 +20,7 @@ function warnAboutReceivingStore() {
 
 export default class Provider extends Component {
   getChildContext() {
-    return { store: this.store }
+    return { store: this.store, react15CompatibilityMode: this.props.react15CompatibilityMode }
```

Comment:
Provider is where you set the default, so you don't have to change it on every single connected component when you want the new behavior.